packages/core/src/TxMetadata/cip20.ts

```diff
@@ -0,0 +1,56 @@
+import { StringUtils } from '@cardano-sdk/util';
+import { TxMetadata } from '../Cardano';
+
+export const CIP_20_METADATUM_LABEL = 674n;
+
+export enum MessageValidationFailure {
+  wrongType = 'WrongType',
+  oversize = 'Oversize'
+}
+
+export type MessageValidationResult = {
+  valid: boolean;
+  failure?: MessageValidationFailure;
+};
+
+export type CIP20TxMetadataMessage = string;
+
+export type CIP20TxMetadataArgs = {
+  // An array of message strings, limited to 64 bytes each
+  messages: CIP20TxMetadataMessage[];
```

Comment:
**(not a change req)** A higher level util that splits a single potentially long message into an array of 64 byte lines would be nice too

Reply:
Good point. That would cover Lace's current needs so we can replace the untested code with this

Reply:
9e729201b5b77a2b3ecb81d832f6b636eb4f5bf4 + 8e217413d4967115d02f4075c936d43d56d2d858